File: module-server/src/main/java/de/twomartens/wahlrecht/model/db/Election.java
package de.twomartens.wahlrecht.model.db;

import de.twomartens.wahlrecht.model.dto.v1.VotingThreshold;
import java.time.LocalDate;
import java.util.Collection;
import java.util.Date;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.FieldDefaults;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
@Getter
@Setter
@Builder
@EqualsAndHashCode
@ToString
@FieldDefaults(level = AccessLevel.PRIVATE)
@NoArgsConstructor(force = true, access = AccessLevel.PRIVATE)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class Election {

  @Id
  ObjectId id;
  @CreatedDate
  Date created;
  @LastModifiedDate
  Date lastModified;

  @Indexed(unique = true)
  String name;
  LocalDate day;
  VotingThreshold votingThreshold;
  int totalNumberOfSeats;
  Collection<Constituency> constituencies;
}

File: module-server/src/main/kotlin/de/twomartens/wahlrecht/model/db/Election.kt
package de.twomartens.wahlrecht.model.db

import de.twomartens.wahlrecht.model.dto.v1.VotingThreshold
import org.bson.types.ObjectId
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.time.Instant
import java.time.LocalDate

@Document
data class Election(
    @Indexed(unique = true) var name: String,
    var day: LocalDate,
    var votingThreshold: VotingThreshold,
    var totalNumberOfSeats: Int,
    var constituencies: Collection<Constituency>
) {
    @Id
    var id: ObjectId = ObjectId()

    @CreatedDate
    lateinit var created: Instant

    @LastModifiedDate
    lateinit var lastModified: Instant

}
